libraries/chef_ingredient_provider.rb

```diff
@@ -16,58 +16,82 @@
 #
 
 require_relative './helpers'
+require_relative './debian_handler'
+require_relative './rhel_handler'
+require_relative './omnitruck_handler'
 
 class Chef
   class Provider
     class ChefIngredient < Chef::Provider::LWRPBase
       provides :chef_ingredient
-      # for include_recipe
+      use_inline_resources
```

Comment:
Only if you're using the new-style providers embedded in the resources which this isn't....